pynq/lib/dma.py

```diff
@@ -28,20 +28,17 @@
 #   ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 from pynq import DefaultIP
-import os
-import cffi
-import functools
-import signal
-import numpy as np
 import warnings
-from pynq.ps import CPU_ARCH_IS_SUPPORTED, CPU_ARCH
 
 
 __author__ = 'Peter Ogden, Anurag Dubey'
 __copyright__ = 'Copyright 2017, Xilinx'
 __email__ = 'pynq_support@xilinx.com'
 
 
+DEFAULT_C_SG_LENGTH_WIDTH = 14
```

Comment:
My one worry is that this will break any designs out there that don't have a .hwh file and have a length register set > 14 bits. I feel we should throw a warning but not an exception if an over-length buffer is used and no description is available.